Added new case insensitive matcher

herbrandson opened the pull request:
This adds the ability to have case insensitive matching for routes. This is important for projects that need to migrate from other load balancers that are case insensitive.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/fabiolb/fabio?pullRequest=553) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/fabiolb/fabio?pullRequest=553) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/fabiolb/fabio?pullRequest=553) it.</sub>

herbrandson commented on route/matcher.go:
```diff
@@ -12,6 +12,7 @@ type matcher func(uri string, r *Route) bool
 var Matcher = map[string]matcher{
 	"prefix": prefixMatcher,
 	"glob":   globMatcher,
+	"noCase": noCaseMatcher,
```
I feel like this might be a terrible name, but it's the best I could come up with this evening. I'm open to suggestions

magiconair replied:
Maybe `nocase`

herbrandson commented:
I've never done GoLang before, but this seemed pretty straight forward based on the other examples. Let me know if I missed anything.

herbrandson commented on route/matcher.go:
```diff
@@ -23,3 +24,10 @@ func prefixMatcher(uri string, r *Route) bool {
+func noCaseMatcher(uri string, r *Route) bool {
+	lowerURI := strings.ToLower(uri)
+	lowerPath := strings.ToLower(r.Path)
+	return strings.HasPrefix(lowerURI, lowerPath)
```
After looking at this again, I think this should be a case insensitive _prefix_ match. Originally I was using `EqualFold`. Thoughts?

magiconair replied:
```go
// todo(fs): if this turns out to be a performance issue we should cache
// todo(fs): strings.ToLower(r.Path) in r.PathLower
return strings.HasPrefix(strings.ToLower(uri), strings.ToLower(r.Path))
```

magiconair commented on config/load.go:
```diff
@@ -247,7 +247,7 @@ func load(cmdline, environ, envprefix []string, props *properties.Properties) (c
 	}
 
-	if cfg.Proxy.Matcher != "prefix" && cfg.Proxy.Matcher != "glob" && cfg.Proxy.Matcher != "noCase" {
+	if cfg.Proxy.Matcher != "prefix" && cfg.Proxy.Matcher != "glob" && cfg.Proxy.Matcher != "nocase" {
```
With respect to the case-insensitive prefix matching I think `iprefix` is a better name.

magiconair commented on fabio.properties:
```diff
@@ -297,6 +297,7 @@
 #
 # prefix: prefix matching
 # glob:  glob matching
+# nocase: matches using a case insensitive test
```
```
# iprefix: case-insensitive prefix matching
```

herbrandson commented:
I believe I've addressed all of your comments. I just wanted to make sure there wasn't something I missed that you're still waiting on. Are there any other changes you'd like to see on this?

herbrandson commented:
Any update on this? We really need this in order to migrate to fabio

magiconair commented:
Thank you and sorry that it took so long!
